staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -73,6 +73,10 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
+		switch err.(type) {
+		case x509.UnknownAuthorityError:
```

Comment:
> I think that this is an error condition. I would not expect to fall back to anonymous auth, but to return a 401

The chain looks like this:

```
UnionFailOnError
    AuthenticatedGroupAdder
        UnionDon'tFailOnError
            RequestHeader - with CABundle1
            ClientCert - with CABundle2
            TokenAccessReview
    Anonymous
```
If the client sent a client cert for a different authority, it seems like it ought to just say, "not for me" and let the next authenticator have a try.